### PR TITLE
Feat/3584 replace collapse menu to drop-down and update all drop-down menus

### PR DESCRIPTION
**Pull request by asf-algoo**
#3584 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend/i18next.scanner/en/translation.json [hunk 1]
```diff
@@ -165,8 +165,6 @@
   "(protocole WebDAV)": "(WebDAV protocol)",
   "Administration": "Administration",
   "Agendas": "Agendas",
-  "See space": "See space",
```
Comment (grignards) [inline frontend/i18next.scanner/en/translation.json:168]: "Show space" would perhaps be better?

File: frontend/i18next.scanner/fr/translation.json [hunk 2]
```diff
@@ -166,8 +166,6 @@
   "(protocole WebDAV)": "(protocole WebDAV)",
   "Administration": "Administration",
   "Agendas": "Agendas",
-  "See space": "Voir l'espace",
```
Comment (grignards) [inline frontend/i18next.scanner/fr/translation.json:169]: "Afficher l'espace" if "Show Space" in english.

File: frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+const DropdownMenu = props => {
+  return (
+    <div className='dropdown'>
+      <button
+        aria-expanded='false'
+        aria-haspopup='true'
+        className={classnames(
+          'btn dropdown-toggle',
+          props.isButton ? 'nohover primaryColorBorder' : 'transparentButton',
+          props.customClassButton
```
Comment (grignards) [inline frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx:15]: `customClassButton`?

File: frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        )}
+        data-cy={props.dataCyButton}
```
Comment (grignards) [inline frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx:17]: Same, `buttonDataCy`?

File: frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+export default DropdownMenu
+
+DropdownMenu.propTypes = {
+  buttonIcon: PropTypes.string,
+  buttonImage: PropTypes.string,
+  buttonLabel: PropTypes.string,
+  buttonTooltip: PropTypes.string,
+  customClassButton: PropTypes.string,
+  customClassItem: PropTypes.string,
+  customClassMenu: PropTypes.string,
```
Comment (grignards) [inline frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx:54]:
```suggestion
  itemCustomClass: PropTypes.string,
  menuCustomClass: PropTypes.string,
```

**Review by grignards — APPROVED** [review]

File: frontend/i18next.scanner/en/translation.json (hunk 1 above)
Comment (gbonaspetti) [reply frontend/i18next.scanner/en/translation.json:168]: I deleted this translations, I put "Actions" as tooltip (like in the action menu from All contents).

File: frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx (hunk 3 above)
Comment (gbonaspetti) [reply frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx:15]: as it is a generic component, the calling component may want to apply some specific property (for example, a background color) to the button that toggles the drop-down menu...

Comment (gbonaspetti) [reply frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx]: after reading the other comments I understood that you might be asking about the name. I update to buttonCustomClass :smile:

**Comment by gbonaspetti** [comment]
For quality tests:
- It is normal that the spaces avatars are a bit buggy, they will be replaced in issue https://github.com/tracim/tracim/issues/3579, so no development was done to fix them here.
- The actions menu at sidebar shows only when holver
- The menus that were updated were:
     - Administration at header
     - Language at header
     - Profile at header
     - Status at contents
     - Role at dashboard advanced
     - Crate content at all contents
     - Create content in folder at all contents (in a folder)
     - Actions at all contents (in all types of content)
     - Actions at sidebar
- Please, test with the scroll at the sidebar still works fine

File: frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        {props.buttonLabel && <span>{props.buttonLabel}</span>}
+      </button>
+
+      <div
+        aria-labelledby='dropdownMenuButton'
+        className={classnames('dropdownMenu dropdown-menu', props.menuCustomClass)}
+      >
+        {props.children.length > 1
+          ? props.children.map(child => <DropdownMenuItem customClass={props.itemCustomClass}> {child} </DropdownMenuItem>)
+          : <DropdownMenuItem customClass={props.itemCustomClass}> {props.children} </DropdownMenuItem>
```
Comment (raphj) [inline frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx:37]: You probably need a `key` attribute here. Linting is disabled in frontend_lib IIRC, we can see how you can run them if you need.

**Review by raphj — APPROVED** [review]

File: frontend/src/component/Header/MenuActionListItem/DropdownLang.jsx [hunk 7]
```diff
@@ -1,39 +1,32 @@
...
+        buttonLabel={activeLang.label}
+        buttonCustomClass='dropdownlang__dropdown__btnlanguage outlineTextBtn'
+        menuCustomClass='dropdownlang__dropdown__subdropdown'
+        isButton
+      >
+        {props.langList.filter(lang => lang.id !== props.langActiveId).map(lang =>
+          <button
+            className='transparentButton'
+            data-cy={lang.id}
+            onClick={() => props.onChangeLang(lang.id)}
```
[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend/src/component/Header/MenuActionListItem/DropdownLang.jsx:22]: hahaha not linked to this pr :wink:

File: frontend/src/component/Header/MenuActionListItem/DropdownLang.styl [hunk 8]
```diff
@@ -1,28 +1,13 @@
...
-        font-size 14px
-        cursor pointer
-        &:hover:not(:active)
-          background-color grey-hover
         &__flag
-          width 25px
-          height 18px
-          vertical-align text-top
+          width 20px
+          height 15px
```
[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend/src/component/Header/MenuActionListItem/DropdownLang.styl:11]: I agree, but I think we need an issue to do a big refactor of "hard-coded" values

File: frontend/src/component/Sidebar/WorkspaceListItem.jsx [hunk 9]
```diff
@@ -52,14 +58,18 @@ class WorkspaceListItem extends React.Component {
     return props.label.substring(0, 2).toUpperCase()
   }
 
+  handleMouseHoverAtItem = () => this.setState(prev => ({ showDropdownMenuButton: !prev.showDropdownMenuButton }))
```
[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend/src/component/Sidebar/WorkspaceListItem.jsx:61]: This function is called when we "leave" and when we "enter" the item

File: frontend/src/component/Header/MenuActionListItem/DropdownLang.styl (hunk 8 above)
[the remark this answers is not in the document]
Comment (raphj) [reply frontend/src/component/Header/MenuActionListItem/DropdownLang.styl:11]: Yes, but my opinion is that we can start doing it for new values and fixes. We may need to discuss the strategy.

File: frontend/src/component/Sidebar/WorkspaceListItem.jsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (raphj) [reply frontend/src/component/Sidebar/WorkspaceListItem.jsx:61]: Yes, its name should reflect this I think

[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend/src/component/Sidebar/WorkspaceListItem.jsx:61]: You don't think that "handle mouse hover" reflect it ? The hover is the enter/leave for me...

File: frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx (hunk 6 above)
Comment (gbonaspetti) [reply frontend_lib/src/component/DropdownMenu/DropdownMenu.jsx:37]: Updated.

File: frontend/src/component/Header/MenuActionListItem/DropdownLang.styl (hunk 8 above)
[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend/src/component/Header/MenuActionListItem/DropdownLang.styl:11]: Yes, in this case I didn't add the values, I changed the name of the class, but if you see the deleted lines there is this values...

File: frontend/src/component/Sidebar/WorkspaceListItem.jsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (raphj) [reply frontend/src/component/Sidebar/WorkspaceListItem.jsx:61]: In CSS, `:hover` is when the cursor is on the element. In JS, mouseover (without H) is when you enter the element. It's not the same but it looks extremely similar too me I think the name of the function could be a tad more explicit.

However, it's probably fine as is and you can keep it like this. As you whish.

**Review by raphj — APPROVED** [review]

File: frontend/src/component/Sidebar/WorkspaceListItem.jsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend/src/component/Sidebar/WorkspaceListItem.jsx:61]: maybe just `handleItemHover`?

[the remark this answers is not in the document]
Comment (raphj) [reply frontend/src/component/Sidebar/WorkspaceListItem.jsx:61]: I'd go with handleItemEnterAndLeave but I agree it's a bit verbose. handleItemHover is fine, and better than handleMouseHoverAtItem.

[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend/src/component/Sidebar/WorkspaceListItem.jsx]: Updated

**Review by raphj — APPROVED** [review]

**Pull request merged**